### DISCUSSION TITLE
Where can I find the parameters for converting to AVIF?

### DISCUSSION THREAD

**Discussion by oG14108149527Exl0Ftr7sY** [ImageMagick/ImageMagick#7821] [Help]
### ImageMagick version

7.1.1-41

### Operating system, version and so on

macOS Sequoia 15.1.1

### Description

Use this for batch conversion:

`for image in *.jpg ; do convert -quality 40% -define avif:speed=3 '$image' '${image%.*}raincoat-blue-.avif' ; done
`

Where can I find a description of the parameters that can be used for AVIF conversion, such as:

`-quality 40%
`

`-define avif:speed=3
`

There are many more parameters that can be used.
Where can I find them?
Where can I find a description of how and when to use the parameters?

**Comment by snibgo** [comment]
I strongly suggest using "magick" instead of "convert", which is deprecated.

Where did you find "-define avif:speed=3"? Does it have any effect? I can't find that documented or in the source, and it seems to have no effect. Perhaps you intended "-define heic:speed=3". "heic:" defines are documented at https://imagemagick.org/script/defines.php

For "-quality", see https://imagemagick.org/script/defines.php . However, that documentation is out of date, as that setting affects more formats than just JPEG, MIFF and PNG.

**Comment by oG14108149527Exl0Ftr7sY** [comment]
Hi

Thank you very much for responding so quick and good.




# 1. #
Re:
Where did you find `"-define avif:speed=3"`? 


Answer:
From:

[https://linuxconfig.org/converting-images-to-avif-on-linux-including-jpg-png-and-webp-formats](https://linuxconfig.org/converting-images-to-avif-on-linux-including-jpg-png-and-webp-formats
)

I spent a few hours Googling to find out about converting to AVIF.
This is one of the better "how to" pages.



# 2. #
"`-define avif:speed=3`" seems to have no effect.



# 3. #
When converting to AVIF using Google Squoosh:

[https://squoosh.app/editor](https://squoosh.app/editor)

you have the options:

"Reduce palette"

and

"Effort"

Under "Advanced Settings" you have another nine settings to optimize AVIF. A total of 11 different settings to optimize AVIF.


# 4. #
## I am searching for a "Howto" and/or other resources to use the same 11 settings with ImageMagick. ##


# 5. #
Been using WEBP for many years. Now it's time to switch to AVIF because:

A.
All major browsers support AVIF.


B.
Google "Structured Data" supports AVIF:

[https://developers.google.com/search/docs/appearance/google-images
](https://developers.google.com/search/docs/appearance/google-images
)

C.
"Adobe Lightroom Classic and Adobe Camera Raw support AVIF.


D.
Cloudflare.com supports AVIF.


F.
WordPress supports AVIF.

[https://www.stanventures.com/news/avif-vs-webp-seo-recommended-712/](https://www.stanventures.com/news/avif-vs-webp-seo-recommended-712/)

**Comment by oG14108149527Exl0Ftr7sY** [comment]
Friday, August 30, 2024: "Supporting AVIF in Google Search"

[https://developers.google.com/search/blog/2024/08/happy-avifriday](https://developers.google.com/search/blog/2024/08/happy-avifriday)

**Comment by UaE5LeN23DMjGfr** [comment]
Really tho! https://imagemagick.org/script/webp.php exists, but I can't find one for AVIF. That would be a foundational resource for the web, AVIF is free bandwidth

**Reply by fmw42** [reply]
magick -list format

shows me

Format  Module    Mode  Description
AVIF  HEIC      rw+   AV1 Image File Format (1.19.8)

So AVIF is in the HEIC module.
